### ISSUE TITLE
showDialog glitch transparent line / border when container in specific height

### ISSUE THREAD

**Issue by sooxt98** [flutter/flutter#40063]
![image](https://user-images.githubusercontent.com/13378059/64493503-71b24c00-d2b3-11e9-8b8c-3e5993ae1359.png)

``` dart
showDialog(
  context: context,
  builder: (BuildContext context) {
    // return object of type Dialog
    return Center(
      child: Container(
        padding: EdgeInsets.all(8),
        color: Colors.red,
        // height: 200,
        width: 200,
        child: Container(
          child: Column(
            mainAxisSize: MainAxisSize.min,
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              Container(
                height: 20.0,
                width: 200,
                // margin: const EdgeInsets.all(10.0),
                color: Colors.white,
              ),
              Container(
                height: 3,
                width: 200,
                color: Colors.white,
              ),
              Container(
                height: 3,
                width: 200,
                color: Colors.white,
              ),
              Container(
                height: 3,
                width: 200,
                color: Colors.white,
              ),
            ],
          ),
        ),
      ),
    );
  }
);
```

**Comment by HansMuller** [comment]
This looks like a known problem with antialiasing and mapping from logical pixels to device pixels. There's an open issue for a documentation fix: #14288

More info about the problem can be found here:
#14288 (comment)
#17084 (comment)
#15035 (comment)

**Comment by TahaTesser** [comment]
Testing on the latest `dev` channel, i cannot reproduce this issue 

<details> 
<summary>Code Sample</summary> 

``` 
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: Container(
          child: Text('Hello World'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
          child: Icon(Icons.add),
          onPressed: () {
            showDialog(
                context: context,
                builder: (BuildContext context) {
                  // return object of type Dialog
                  return Center(
                    child: Container(
                      padding: EdgeInsets.all(8),
                      color: Colors.red,
                      // height: 200,
                      width: 200,
                      child: Container(
                        child: Column(
                          mainAxisSize: MainAxisSize.min,
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: <Widget>[
                            Container(
                              height: 20.0,
                              width: 200,
                              // margin: const EdgeInsets.all(10.0),
                              color: Colors.white,
                            ),
                            Container(
                              height: 3,
                              width: 200,
                              color: Colors.white,
                            ),
                            Container(
                              height: 3,
                              width: 200,
                              color: Colors.white,
                            ),
                            Container(
                              height: 3,
                              width: 200,
                              color: Colors.white,
                            ),
                          ],
                        ),
                      ),
                    ),
                  );
                });
          }),
    );
  }
}

``` 

</details> 

<img width="1249" alt="Screenshot 2020-05-05 at 7 58 54 PM" src="https://user-images.githubusercontent.com/48603081/81077800-13ee0680-8f0b-11ea-9699-ee212496c147.png">


<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, v1.18.0-8.0.pre, on Mac OS X 10.15.4 19E287, locale
    en-GB)
    • Flutter version 1.18.0-8.0.pre at /Users/taha/Code/flutter_dev
    • Framework revision e0c63cd35e (12 days ago), 2020-04-23 22:29:01 -0400
    • Engine revision d2ec21221e
    • Dart version 2.9.0 (build 2.9.0-3.0.dev b0d35855d8)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.44.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.1

[✓] Connected device (5 available)
    • SM M305F      • 32003c30dc19668f          • android-arm64  • Android 10
      (API 29)
    • Taha’s iPhone • 00008020-001059882212002E • ios            • iOS 13.4.1
    • macOS         • macOS                     • darwin-x64     • Mac OS X
      10.15.4 19E287
    • Web Server    • web-server                • web-javascript • Flutter Tools
    • Chrome        • chrome                    • web-javascript • Google Chrome
      81.0.4044.129

• No issues found!
``` 

</details> 

if you're still experiencing this problem, please create a new issue and provide the information
required in the template.
Thank you
